.github/workflows/rst.yaml

```diff
@@ -24,7 +24,12 @@ jobs:
         run: |
           export PYTHONPATH=$PYTHONPATH:$PWD
           python ./scripts/rst.py
-          
+
+      - name: Check for changes
+        id: changes
+        run: |
+          git diff --quiet && echo 'No changes to commit' || echo '::set-output name=changed::true'
+
       - name: Save changes
```

Comment:
This checks for changes but you don't use the check as conditional logic for saving. You'd need something like

```sh
- name: Save changes
        if: steps.changes.outputs.changed == 'true'
```
and

```sh
- name: push changes
        if: steps.changes.outputs.changed == 'true'
```

But set-output will complain (deprecated) so you should echo to the env:

```sh
|| echo 'changed=true' >> $GITHUB_ENV
```
(this is just the bit from `||`)

which then would look like:

```sh
- name: Save changes
        if: env.changed == 'true'
```